samples/AWS/Frontend/Program.cs

```diff
@@ -0,0 +1,47 @@
+using Amazon.DynamoDBv2;
+using Amazon.Runtime;
+using Frontend.Components;
+using Frontend.Data;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.AddServiceDefaults();
+
+// TODO: This logic can be simplified to always use the example credentials once I'm able to adjust the
+// COMMAND used to startup the container. Then I can set the DynamoDB local database to not include the
+// Access Key ID as part of the file name.
+AWSCredentials credentials;
+try
+{
+    credentials = FallbackCredentialsFactory.GetCredentials();
+}
+catch
+{
+    // DynamoDB local just needs something that looks like credentials. These are not real credentials as you call with the "EXAMPLE" suffix.
+    credentials = new BasicAWSCredentials("AKIAIOSFODNN7EXAMPLE", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY");
+}
+
+builder.Services.AddSingleton<IAmazonDynamoDB>(new AmazonDynamoDBClient(credentials));
```

Comment:
I agree we need components for adding the service clients like DynamoDB. That is a general problem for all AWS service clients that I would like to solve in a separate PR.

The `FallbackCredentialsFactory.GetCredentials()` line above this is doing what you are saying about workload-based identity. I put this fallback of using the dummy credentials so people who didn't have an AWS setup could still test the sample. I'm guessing few on the .NET team have AWS credential setups :smile:. But I do admit this will not work once we add more AWS components and those require real credentials.

So my question back to the team, how much do you want to have the samples be F5 debuggable even though people might not have all of the normal prerequisites.

Reply:
I think when we are talking about cloud specific components like we are here, we don't have to be as strict with the F5 debuggable without doing some local prep work first. But we should explain what people need to do (maybe in a README) to get the sample running assuming they can meet those requirements.

For the eShopLite example we have a dependency on service bus, but its a soft dependency ... the sample will still run without it. But eShopLite is our main sandpit when we are trying things out in the repo before we build the necessary test harnesses around it :)

Reply:
I hadn't fleshed out the idea but wondered if provisioners needs to be a more first class concept. When they are registered they they could show up in the dashboard and the user could decide when to invoke the provisioner should be invoked instead of being something that is always done as part F5. I'm worried about disrupting the F5 experience especially after the initial F5 session checking to see if anything needs to be done. Being in the dashboard would allow us to have the provisioner interface have a tear down method that could be invoked from the dashboard making it easier to tear things down. If you all think this idea warrants further discussion I can open a top level discussion on the topic.